champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/axe/MagneticAxe.java

```diff
@@ -0,0 +1,455 @@
+import me.mykindos.betterpvp.core.combat.events.CustomDamageEvent;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilDamage;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import me.mykindos.betterpvp.core.utilities.UtilVelocity;
+import org.bukkit.Effect;
+import org.bukkit.FluidCollisionMode;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.Particle;
+import org.bukkit.Sound;
+import org.bukkit.entity.ItemDisplay;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.event.player.PlayerQuitEvent;
+import org.bukkit.event.player.PlayerTeleportEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.util.NumberConversions;
+import org.bukkit.util.RayTraceResult;
+import org.bukkit.util.Transformation;
+import org.bukkit.util.Vector;
+import org.joml.AxisAngle4f;
+import org.joml.Vector3f;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+@Singleton
+@BPvPListener
+public class MagneticAxe extends Skill implements InteractSkill, Listener, CooldownSkill, OffensiveSkill, DamageSkill {
+
+    private final WeakHashMap<Player, List<AxeData>> axeDataMap = new WeakHashMap<>();
+    private final WeakHashMap<Player, List<AxeData>> returningAxes = new WeakHashMap<>();
+
+    private double baseDamage;
+    private double damageIncreasePerLevel;
+    private double duration;
+    private double hitboxSize;
+    private double xSize;
+    private double ySize;
+    private double zSize;
+    private double magnitude;
+    private double gravity;
+    private double yawOffset;
+    private double dragConstant;
+    private int rotations;
+    private double initialAngle;
+    private double blocksPerSecond;
+    private double maxVelocity;
+
+    @Inject
+    public MagneticAxe(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Magnetic Axe";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
+                "Right click with an Axe to activate",
+                "",
+                "Throw your axe, dealing " + getValueString(this::getDamage, level),
+                "damage to anything you hit and",
+                "pulling your axe back to you",
```

Comment:
could probably improve this a bit, the whole sentence doesnt make much sense

Reply:
fixed